src/GameLogic/Player.cs

```diff
@@ -922,6 +956,35 @@ public async ValueTask WarpToAsync(ExitGate gate)
         // Then ClientReadyAfterMapChange is called.
     }
 
+    /// <summary>
+    /// Moves the player to the specified map and coordinates.
+    /// </summary>
+    /// <param name="map">The map to which the player should be warped.</param>
+    /// <param name="position">The coordinates on the target map.</param>
+    public async ValueTask WarpToMapAsync(GameMap map, Point position)
```

Comment:
It looks similar to WarpToAsync. Can you refactor WarpToAsync to call this method?
Also, WarpToAsync calls PlaceAtGate which also moves the summoned monster to the new coordinates. We'll need that for the summon party skill as well.
Alternatively: refactor TeleportToMapAsync to smth. like TeleportToGateAsync. Then remove WarpToMapAsync, define a temporary ExitGate object for the area of the dark lord and do something like player.TeleportToGateAsybc -> WarpToAsync

Reply:
Yeah, I actually copied this from WarpToAsync. But instead of calling PlaceAtGate(), I added them manually here which is also what PlaceAtGate() does (except it does not randomize anymore).
```
        this.SelectedCharacter!.CurrentMap = map.Definition;
        this.SelectedCharacter.PositionX = position.X;
        this.SelectedCharacter.PositionY = position.Y;
```
The problem I have with PlaceAtGate() is that it will try to randomize another set of X,Y when I already have a prepared X,Y beforehand.

Reply:
Anyway, I just realized I could make X1 = X2 Y1 = Y2 so the randomization in PlaceAtGate won't matter.

Reply:
Updated as suggested.